docs/madr/decisions/049-policy-on-namespace.md

```diff
@@ -0,0 +1,613 @@
+3. **[new step]** Policy in any custom namespace is more specific than any policy in the `kuma-system` namespace.
+   When both policies either from custom namespaces or from `kuma-system`, go to the next step.
+4. **[new step]** Policy from local zone is more specific than a policy from another zone. 
+   When policies have the same zone, go to the next step.
+5. **[new step]** Policy from local namespace is more specific than a policy from another namespace.
+   When namespaces are equal, go to the next step.
+6. The policy is more specific if it's name is lexicographically less than other policy's name ("aaa" < "bbb" so that "
+   aaa" policy is more specific)
+
+### Cross-resource references
+
+When a policy in `zone-1` references a resource (MeshService or MeshHTTPRoute), 
+the reference should work after policy is synced to `zone-2`.
+
+Using `to[].targetRef.labels` is a reliable way to reference the resource, because labels are constant across zones. 
+For example, the label `kuma.io/display-name` has the same value no matter what zone or global we're using to read the resource. 
+That's why targetRefs like
+
+```yaml
+to:
+  - targetRef:
+      kind: MeshService
+      labels:
+        kuma.io/display-name: backend
+```
+are going to work even when policy is synced to another zone.
+The downside of this approach is that multiple MeshServices can be selected, 
+because we didn't specify `kuma.io/zone` and `k8s.kuma.io/namespace` labels.
+
+Another way to reference the resource is by using `name` and `namespace` fields:
+
+```yaml
+to:
+  - targetRef:
+      kind: MeshService
+      name: backend
+      namespace: backend-ns
+```
+
+but we have to implement "smart" de-referencing when the policy is synced to another zone.
+The code that performs de-referencing should take into account `kuma.io/zone`, `k8s.kuma.io/namespace` labels of the policy
+and find the single correct resource that was referenced. For example:
+
+```yaml
+# zone-1, producer-timeout references backend service
+kind: MeshService
+metadata:
+  name: backend
+  namespace: backend-ns
+---
+kind: MeshTimeout
+metadata:
+  name: producer-timeout
+  namespace: backend-ns
+spec:
+  to:
+    - targetRef:
+        kind: MeshService
+        name: backend
+```
+
+these resources are synced to `zone-2`:
+
+```yaml
+# zone-2, producer-timeout references backend service
+kind: MeshService
+metadata:
+  name: backend-w3r3jwi90l8dyuix
+  namespace: kuma-system
+  labels:
+    kuma.io/zone: zone-1
+    k8s.kuma.io/namespace: backend-ns
+    kuma.io/display-name: backend
+---
+kind: MeshTimeout
+metadata:
+  name: producer-timeout-qwea89uy739jdgmu
+  namespace: kuma-system
+  labels:
+    kuma.io/zone: zone-1
+    k8s.kuma.io/namespace: backend-ns
+    kuma.io/display-name: producer-timeout
+spec:
+  to:
+    - targetRef:
+        kind: MeshService
+        name: backend # based on labels we should be able to de-reference this to 'backend-w3r3jwi90l8dyuix'
```

Comment:
can't we replace target refs before sync from global to zone?

Reply:
Right, replacing name with hashed name seems like a better option